internal/adapters/consul/http.go

```diff
@@ -219,18 +221,17 @@ func hostsKey(hosts ...string) string {
 	return strconv.FormatUint(uint64(hostsHash.Sum32()), 16)
 }
 
+// compareHTTPRules implements some of the order of precedence for routes specified by the K8s Gateway API spec.
+// https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.HTTPRouteRule
+//
+// Ordering prefers matches based on the largest number of:
+//
+//   1. characters in a matching non-wildcard hostname
+//   2. characters in a matching hostname
+//   3. characters in a matching path
+//   4. header matches
+//   5. query param matches
```

Comment:
Added a link to the spec where it covers this order of precedence. I also noticed that we don't apply the first and second comparisons here. I'm assuming they're applied elsewhere in the codebase?

Reply:
Nope, it's all internal to `envoy` and AFAIK we have 0 control over it (though envoy's prioritization does match domains this way). I've never found documentation on this, but here are some links that we might consider adding as part of the comment:

- Comment in [Istio](https://github.com/istio/istio/issues/35826#issuecomment-957184380) repo
- Comment in [envoy](https://github.com/envoyproxy/envoy/issues/18874#issuecomment-958430741) repo
- I'm pretty sure this is the [matching code in envoy](https://github.com/envoyproxy/envoy/blob/5c4d4bd957f9402eca80bef82e7cc3ae714e04b4/source/common/router/config_impl.cc#L1645-L1682)

Reply:
Thanks! I added the third link which seemed the most directly related and least ephemeral of the three.